ls1intum/Artemis
# `Assessment`: Allow instructors to select a decimal separator before downloading course scores as csv

### Checklist
#### General


#### Client

### Motivation and Context


When course scores get downloaded, the decimal separator used for the export depends on the language set by the user in Artemis, which is not very transparent for the user. This can also result in problems when the user set a different language in Artemis than their system's setting and imports it into Excel, which than interprets values differently (e.g. Artemis: 'en', User system: 'de' -> Generated csv contains numbers in the format 1.4 -> Excel interprets this as a date (1. April)).

### Description

The user now can specify which decimal separator should be used before the export. 

### Steps for Testing

Prerequisites:

- 1 Course
- 1 Student
- 1 Exercise, on which the student participated

1. Log in to Artemis
2. Navigate to Course Administration, select the course, then Scores
3. Click the `Export as CSV` button
4. Select a decimal separator that should be used for generating the csv file
5. Click `Export`
6. View the downloaded csv file in a text editor and make sure the previously selected option was applied

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-scores-csv-row-builder.ts | 100% | 100% |
| course-scores.component.ts | 73.84% | 94.69% |
| csv-export-modal.component.ts | 87.5% | 96.96% |

### Screenshots


![Export-Chrome](https://user-images.githubusercontent.com/9954674/162969690-ba7aa49a-6aa8-4bdd-8b8e-70c25e2fc6dc.gif)
